### PR TITLE
Alerting: Fix RegExp matchers in frontend for Silences and other previews.

### PR DESCRIPTION
**Pull request by joeblubaugh**
The `regex` and `notRegex` matcher operators were not properly matching
against supplied values. If using an unknown string instead of a known
RegExp array like `/[a-b]*/`, then a new RegExp must be created before
calling `match()`

### CODE REVIEW

**Review by peterholmberg — APPROVED** [review]
This looks good to me, only had a comment regarding the tests

File: public/app/features/alerting/unified/utils/matchers.ts [hunk 1]
```diff
@@ -41,29 +41,44 @@ export const findAlertInstancesWithMatchers = (
...
-      if (matcher.operator === MatcherOperator.equal) {
-        return matcher.name === key && matcher.value === value;
-      }
-      if (matcher.operator === MatcherOperator.notEqual) {
-        return matcher.name === key && matcher.value !== value;
-      }
-      if (matcher.operator === MatcherOperator.regex) {
-        return matcher.name === key && matcher.value.match(value);
+      if (matcher.name !== key) {
+        return false;
```
Comment (konrad147) [inline public/app/features/alerting/unified/utils/matchers.ts:61]: ❤️

**Review by konrad147 — APPROVED** [review]
Nice!

**Comment by joeblubaugh** [comment]
Executive decision: I'm not going to backport to 8.4 and 8.3 - the codebase has changed too much for me to be confident about doing it correctly.

**Pull request merged**